### DISCUSSION TITLE
What is the deps array referring to in form.register? (It doesn't appear to work as it described)

### DISCUSSION THREAD

**Discussion by chcoomilk** [react-hook-form/react-hook-form#9547] [Q&A]
![image](https://user-images.githubusercontent.com/66551661/206362295-08a2508a-e92c-49e0-a61c-8d337fadb407.png)

Hi, I'm trying to figure out how to work with the deps array when form.register()-ing a field. In the docs, there are two strings 'inputA', 'inputB' in the deps array, but I don't what it is referring to. I tried putting an id on a different input field but it didn't work.

**Comment by Moshyfawn** [comment]
The `deps` array accepts names of other registered or controlled form inputs. So, all the other input names you've passed to `register` or `Controller` / `useController`

**Reply by chcoomilk** [reply]
It appears that I have misunderstood what the `deps` is for. I still couldn't trigger the required validation in name field. Or did I do something wrong, I still couldn't trigger the required name validation whenever I clicked at the show age check input or putting valid and invalid numbers in the age field.

Here's the code sandbox for your reference  [https://codesandbox.io/s/react-hook-form-watch-v7-forked-xb4wgp?file=%2Fsrc%2Findex.js](https://codesandbox.io/s/react-hook-form-watch-v7-forked-xb4wgp?file=%2Fsrc%2Findex.js)

**Comment by chcoomilk** [comment]
While using useEffect() might work. The app will break if I tried to refactor the code, so at the first line of the code, `useForm` changed to `useFormContext`
```
const form = useForm<Fields>();
const watch = form.watch();

useEffect(() => {
    form.trigger("duration.second");
}, [form, watch.duration?.day, watch.duration?.hour, watch.duration?.minute]);
```

**Reply by sKopheK** [reply]
i understood `deps` the same way you did and ended up with the same workaround i wanted to avoid :/
